spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/DynamicCassandraPorts.scala

```diff
@@ -0,0 +1,90 @@
+package com.datastax.spark.connector.embedded
+
+import java.net.{ConnectException, Socket}
+import java.nio.file.{FileAlreadyExistsException, Files, Paths}
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+
+import org.apache.spark.Logging
+
+case class DynamicCassandraPorts(basePort: Int) extends CassandraPorts {
+
+  import DynamicCassandraPorts._
+
+  def getRpcPort(index: Int): Int = basePort + RPC_OFFSET + index
+
+  def getJmxPort(index: Int): Int = basePort + JMX_OFFSET + index
+
+  def getSslStoragePort(index: Int): Int = basePort + SSL_STORAGE_OFFSET + index
+
+  def getStoragePort(index: Int): Int = basePort + STORAGE_OFFSET + index
+
+  def release(): Unit = DynamicCassandraPorts.releaseBasePort(basePort)
+}
+
+
+object DynamicCassandraPorts extends Logging {
+  val MaxInstances = 5
+
+  val RPC_OFFSET = 0
+  val JMX_OFFSET = MaxInstances
+  val SSL_STORAGE_OFFSET = MaxInstances * 2
+  val STORAGE_OFFSET = MaxInstances * 3
+
+  private lazy val basePath = Paths.get(sys.props("baseDir"), "target", "ports")
+
+  private lazy val basePorts = 10000 until 20000 by 50
+
+  private def portLockFile(port: Int) = basePath.resolve(s"port_$port.lock")
+
+  def apply(): DynamicCassandraPorts = DynamicCassandraPorts(chooseBasePort())
+
+  def verifyRange(host: String, ports: Seq[Int]): Boolean = {
```

Comment:
Minor nitpick: The method name does not make it obvious what the result means. It has no side effect, so using a verb is also not needed. `areAllPortsAvailable`?